Implemented App Intro

Haid-Faiz opened the pull request:
Fixes #46 

Changes: Implemented App intro using App intro library... and old material intro features are now showing in app intro.


Screenshots for the change:

<table>
<tr>

<td>
<img src = "https://user-images.githubusercontent.com/56159740/99914809-fc029680-2d25-11eb-9793-5dfb04332695.jpg" height = "480" width="250">
</td>

<td>
<img src = "https://user-images.githubusercontent.com/56159740/99933888-4dd90a00-2d82-11eb-9d2f-d22a38f2d6f8.jpg" height = "480" width="250">
</td>

<td>
<img src = "https://user-images.githubusercontent.com/56159740/99914839-1f2d4600-2d26-11eb-9d0b-73d90ade62c3.jpg" height = "480" width="250">
</td>

</tr>
</table>

Haid-Faiz commented:
@rob729 have a look... ktlint check fixed

rob729 commented on app/src/main/java/com/rob729/quiethours/Activity/IntroScreen.kt:
```diff
@@ -0,0 +1,52 @@
+
+class IntroScreen : AppIntro() {
+
+    lateinit var sharedPreferences: SharedPreferences
```
StoreSession is being used for storing values in shared preferences.
Used it for the same.

rob729 commented on app/src/main/java/com/rob729/quiethours/Activity/SplashScreen.kt:
```diff
@@ -24,4 +25,13 @@ class SplashScreen : AppCompatActivity() {
     }
+
+    fun checkAppIntro() {
+        val sharedPreferences = getSharedPreferences("AppIntroCheck", MODE_PRIVATE)
```
same comment as above.

rob729 commented on app/src/main/java/com/rob729/quiethours/Activity/SplashScreen.kt:
```diff
@@ -16,6 +16,7 @@ class SplashScreen : AppCompatActivity() {
         startActivity(intent)
         overridePendingTransition(R.anim.fade_in, R.anim.fade_out)
         checkFirstRun()
+        checkAppIntro()
```
There is already a startActivity() present above checkFirstRun() and then you are adding similar code in checkAppIntro().
Make the changes for the proper working of the app.

rob729 commented:
<img src = "https://user-images.githubusercontent.com/31350501/99883861-fe92bc80-2c4f-11eb-86b3-161288be6823.jpg" width = "200"/>
@Haid-Faiz UI added is not optimized for dark mode as the text is not visible in dark mode.

Haid-Faiz commented:
@rob729 ok i'll fix it...

Haid-Faiz commented:
@rob729  Should i show status bar in app intro...?

rob729 commented:
> @rob729 Should i show status bar in app intro...?

No

rob729 commented:
@Haid-Faiz Also I didn't ask you to add a screenshot of the previous app intro in this one. You had to add the info about the earlier highlighted features.
The current images are not looking good. Maybe you can look into the apps which have implemented the same library to get a better idea. The list of apps using the same library is mentioned in the readme of the library.
Keep in mind to highlight the features which are offered in the current version like swipe to delete, click on the profile to get more details, pause/resume profile. Remove the add profile page, it seems to be too basic to show in app intro.

Just to give a little better idea, in case of the "click on the profile to get more details" feature you can add the screenshot of the details fragment bottom sheet. For the "swipe to delete" feature you can show a sample profile which is swiped towards the right little bit showing the delete icon with the red background.

rob729 commented on app/build.gradle:
```diff
@@ -64,6 +64,9 @@ dependencies {
     def lottieVersion = '3.4.4'
     implementation "com.airbnb.android:lottie:$lottieVersion"
 
+    // App Intro Library
```
@Haid-Faiz Also remove the library of the previous app intro.

rob729 commented on app/src/main/java/com/rob729/quiethours/Activity/IntroScreen.kt:
```diff
@@ -0,0 +1,48 @@
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+//        showStatusBar(true)
```
remove the comment

rob729 commented on app/src/main/java/com/rob729/quiethours/Activity/SplashScreen.kt:
```diff
@@ -12,8 +12,7 @@ class SplashScreen : AppCompatActivity() {
         super.onCreate(savedInstanceState)
-        val intent = Intent(this, MainActivity::class.java)
-        startActivity(intent)
+        checkAppIntro()
```
move the checkFirstRun() above checkAppIntro()

Haid-Faiz replied:
Ok

rob729 commented on app/src/main/java/com/rob729/quiethours/util/StoreSession.kt:
```diff
@@ -61,4 +61,13 @@ object StoreSession {
     fun readLong(key: String): Long {
         return sharedPreferences.getLong(key, 0)
     }
+
```
There is already a function for writeBoolean and readBoolean then why add the below functions?

Haid-Faiz replied:
Ok.. i'll fix it

rob729 commented:
@Haid-Faiz In the second intro screen, crop out the app bar from screenshot

Haid-Faiz commented:
@rob729 ok

rob729 commented:
@Haid-Faiz In the second app intro screen, it should be "Swipe left" instead of slide.
Use the screenshot from the latest version of the app as in the second screenshot the pause/resume toggle is missing.
Also, add the pause/resume profile feature in app intro.
Use roboto_slab font in the app intro.

Haid-Faiz commented:
@rob729 ok

rob729 commented:
@Haid-Faiz The checks are failing. Fix them.

Haid-Faiz commented:
@rob729  Ya.... i will fix.. i'll msg you when everything will completed

Commit:
Implemented App Intro

Haid-Faiz commented:
@rob729 have a look now... & what is this codeClimate check....?? it is showing some similar code warning... How to pass this check ?

rob729 commented:
Click on the details present on the codeclimate check and it will show what are the problems present in the PR.

Haid-Faiz commented:
@rob729  i have fixed 2 issues out of 4 of codeclimate but remaining 2 issues are not issues i think... because it is showing 2 similar block of code have found but actually one block is overridden method onSkipPressed() & second block is onDonePressed() present in IntroScreen Activity. So, now what should i do to fix this

rob729 commented:
> @rob729 i have fixed 2 issues out of 4 of codeclimate but remaining 2 issues are not issues i think... because it is showing 2 similar block of code have found but actually one block is overridden method onSkipPressed() & second block is onDonePressed() present in IntroScreen Activity. So, now what should i do to fix this

Since there is same code in both functions, move that block of code into a function. You can name the function to be navigateToMainFragment().

rob729 commented:
@Haid-Faiz One more thing, in the second app intro screen make it "swipe left a profile to delete it".
Also, make 'Y' capital of you.

Haid-Faiz commented:
> > @rob729 i have fixed 2 issues out of 4 of codeclimate but remaining 2 issues are not issues i think... because it is showing 2 similar block of code have found but actually one block is overridden method onSkipPressed() & second block is onDonePressed() present in IntroScreen Activity. So, now what should i do to fix this
> 
> Since there is same code in both functions, move that block of code into a function. You can name the function to be navigateToMainFragment().

@rob729 Ya, fixed this but read your msg after fixing it.. i mean i named it appIntroClickListeners()

Haid-Faiz commented:
> @Haid-Faiz One more thing, in the second app intro screen make it "swipe left a profile to delete it".
> Also, make 'Y' capital of you.

ok

rob729 commented on app/src/main/java/com/rob729/quiethours/util/AppConstants.kt:
```diff
@@ -9,4 +9,5 @@ object AppConstants {
     val FIRST_BOOT = "FIRST_BOOT"
     val NIGHT_MODE = "nightMode"
     val RINGTONE_MODE = "RINGTONE_MODE"
+    val APP_INTRO_CHECK = "isIntroduced"
```
Make it val APP_INTRO_CHECK = "APP_INTRO_CHECK"

Haid-Faiz replied:
ok

rob729 commented on app/src/main/java/com/rob729/quiethours/Activity/IntroScreen.kt:
```diff
@@ -0,0 +1,49 @@
+        appIntroClickListeners()
+    }
+
+    private fun appIntroClickListeners() {
```
Change function name to navigateToMainFragment()

Haid-Faiz replied:
ok

Haid-Faiz replied:
@rob729 navigateToMainFragment() or navigateToMainActivity()    ...??

rob729 replied:
Sorry, it should be Activity

Haid-Faiz replied:
ok

Haid-Faiz commented:
@rob729  navigateToMainFragment()   OR navigateToMainActivity()

Haid-Faiz commented:
@rob729 Now, have a look

rob729 commented:
give @Haid-Faiz 75 points
